### ISSUE TITLE
bootloader fails to build on Alpine

### ISSUE THREAD

**Issue by DevOpsJon** [pyinstaller/pyinstaller#4196]
* What is the exact problem?

I am attempting to build PyInstaller 3.4 from source in an Alpine Linux container. The  build fails with the following errors: 
```
Waf: Entering directory `/tmp/pyinstaller/bootloader/build/debug'
[ 1/13] Compiling src/pyi_python.c
[ 2/13] Compiling src/pyi_global.c
[ 3/13] Compiling src/pyi_archive.c
[ 4/13] Compiling src/main.c
[ 5/13] Compiling src/pyi_utils.c
[ 6/13] Compiling src/pyi_path.c
[ 7/13] Compiling src/pyi_win32_utils.c
[ 8/13] Compiling src/pyi_pythonlib.c
[ 9/13] Compiling src/pyi_main.c
../../src/pyi_path.c: In function 'pyi_path_join':
../../src/pyi_path.c:143:5: error: 'strncpy' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
     strncpy(result, path1, strlen(path1));
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_path.c:156:9: error: 'strncat' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
         strncat(result, path2, len - 2);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_path.c:152:11: note: length computed here
     len = strlen(path2);
           ^~~~~~~~~~~~~
cc1: all warnings being treated as errors

../../src/pyi_pythonlib.c: In function 'pyi_pylib_start_python':
../../src/pyi_pythonlib.c:469:9: error: 'strncat' specified bound 1 equals source length [-Werror=stringop-overflow=]
         strncat(pypath, PYI_SEPSTR, strlen(PYI_SEPSTR));
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:470:9: error: 'strncat' specified bound 16 equals source length [-Werror=stringop-overflow=]
         strncat(pypath, "base_library.zip", strlen("base_library.zip"));
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:471:9: error: 'strncat' specified bound 1 equals source length [-Werror=stringop-overflow=]
         strncat(pypath, PYI_PATHSEPSTR, strlen(PYI_PATHSEPSTR));
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:468:9: error: 'strncpy' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
         strncpy(pypath, status->mainpath, strlen(status->mainpath));
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:472:9: error: 'strncat' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
         strncat(pypath, status->mainpath, strlen(status->mainpath));
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:464:9: error: 'strncpy' specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
         strncpy(pypath, status->mainpath, strlen(status->mainpath));
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

Waf: Leaving directory `/tmp/pyinstaller/bootloader/build/debug'
Broken revdeps detected on {
	{task 140266198207176: strip run_d -> },
	{task 140266198207064: cprogram main.c.1.o,pyi_archive.c.1.o,pyi_global.c.1.o,pyi_launch.c.1.o,pyi_main.c.1.o,pyi_path.c.1.o,pyi_python.c.1.o,pyi_python27_compat.c.1.o,pyi_pythonlib.c.1.o,pyi_utils.c.1.o,pyi_win32_utils.c.1.o -> run_d}}
The command '/bin/sh -c git clone --depth 1 --single-branch --branch v3.4 https://github.com/pyinstaller/pyinstaller.git /tmp/pyinstaller     && cd /tmp/pyinstaller/bootloader     && python ./waf configure --no-lsb all     && pip install ..     && rm -Rf /tmp/pyinstaller' returned a non-zero code: 1
```

* How to reproduce

1. Clone this repo: https://github.com/six8/pyinstaller-alpine
2. Run build.sh

It doesn't seem to matter what base image I use for alpine for this error.

* Further down the spiral

Originally I ran into this issue: https://github.com/gliderlabs/docker-alpine/issues/48

What would it take to get out of the box support for Alpine and Python 3.7 from pyinstaller so that this workaround isn't needed, and one can simply `pip install pyimage` in an alpine DockerFile?

**Comment by linxon** [comment]
bump

**Comment by linxon** [comment]
```
sh bash 4.4_p23-r1
ld GNU ld (Gentoo 2.31.1 p5) 2.31.1
distcc 3.3.2 x86_64-pc-linux-gnu [disabled]
app-shells/bash:          4.4_p23-r1::gentoo
dev-java/java-config:     2.2.0-r4::gentoo
dev-lang/perl:            5.26.2::gentoo
dev-lang/python:          2.7.15::gentoo, 3.6.5::gentoo
dev-util/cmake:           3.9.6::gentoo
dev-util/pkgconfig:       0.29.2::gentoo
sys-apps/baselayout:      2.6-r1::gentoo
sys-apps/openrc:          0.41.2::gentoo
sys-apps/sandbox:         2.13::gentoo
sys-devel/autoconf:       2.13-r1::gentoo, 2.69-r4::gentoo
sys-devel/automake:       1.11.6-r3::gentoo, 1.15.1-r2::gentoo, 1.16.1-r1::gentoo
sys-devel/binutils:       2.31.1-r4::gentoo
sys-devel/gcc:            8.2.0-r6::gentoo, 8.3.0-r1::gentoo
sys-devel/gcc-config:     2.0::gentoo
sys-devel/libtool:        2.4.6-r3::gentoo
sys-devel/make:           4.2.1-r4::gentoo
sys-kernel/linux-headers: 4.14-r1::gentoo (virtual/os-headers)
sys-libs/glibc:           2.28-r6::gentoo

```

**Comment by rokm** [comment]
Closing this, as bootloader now builds on Alpine (tested with Alpine 3.12.0). The issue in question should have been addressed by commit 22ad4f7200a56da969e9296d8fecf2f0b8368afc (and/or whole PR #4585). If the build problem persists, please open a new issue with updated details.
